packages/context/src/resolution-session.ts

```diff
@@ -366,3 +367,12 @@ export function asResolutionOptions(
   }
   return optionsOrSession ?? {};
 }
+
+/**
+ * Context for resolution
+ */
+export interface ResolutionContext<T = unknown> {
+  context: Context;
+  binding: Readonly<Binding<T>>;
+  options: ResolutionOptions;
```

Comment:
Should we mark all fields as `readonly` to prevent accidental modification?

```ts
export interface ResolutionContext<T = unknown> {
  readonly context: Context;
  readonly binding: Readonly<Binding<T>>;
  readonly options: Readonly<ResolutionOptions>;
}
```